### PR TITLE
Adds payment intent creation/capture tests for valid credid card types (non-3D)

### PR DESCRIPTION
**Pull request by filipefurtad0**
#### What? Why?

- Does not close, but contributes to #11892. See my [reasoning below](https://github.com/openfoodfoundation/openfoodnetwork/pull/11900#issuecomment-1841067498).

Extends existing payment intent validator tests to capture payments from non-3D cards. 

#### What should we test?

- green build.

#### Release notes

Changelog Category (reviewers may add a label for the release notes):

- [ ] User facing changes
- [ ] API changes (V0, V1, DFC or Webhook)
- [x] Technical changes only
- [ ] Feature toggled

The title of the pull request will be included in the release notes.

### CODE REVIEW

**Comment by filipefurtad0** [comment]
One question (hoping not to pull you into my rabbit-hole :hole: :rabbit2:  @openfoodfoundation/developers :-D):
this spec now tests valid/invalid (non-3D) cards. I think it would be feasible to extend it now to 3D cards. This is all green and nice.

But - and now for my question: this is not how we use Stripe on OFN, right?

We use Stripe through [`lib/active_merchant/billing/gateways/stripe_payment_intents_decorator.rb`](https://github.com/openfoodfoundation/openfoodnetwork/blob/master/lib/active_merchant/billing/gateways/stripe_payment_intents_decorator.rb) and I cannot find a spec for this.

So, all the code lines on this PR trigger requests "directly" to Stripe. This is easy to demonstrate: If I just delete the file above, I can still make the specs here pass. So, while it is interesting to see how it works, I don't think I'm really catching anything relevant with this branch.

So I think one way forward could be to write a similar spec as in this PR but for the `stripe_payment_intents_decorator` file. What do you think?

**Comment by mkllnk** [comment]
> write a similar spec as in this PR but for the stripe_payment_intents_decorator file.

You are right that this is a great exercise to get familiar with the testing of Stripe. And while we don't need to test Stripe directly, it's nice to verify that our config and setup works. 

To achieve good coverage, we need to look at all classes that use Stripe and test them. Examples are:

* app/models/spree/gateway/stripe_sca.rb
* app/models/stripe_account.rb
* app/services/stripe_payment_status.rb
* lib/stripe/*


In regards to that decorator, I'm not sure which observable difference it makes. It was supposed to ease the upgrade to newer versions:

* https://github.com/openfoodfoundation/openfoodnetwork/pull/7286
* https://github.com/openfoodfoundation/openfoodnetwork/pull/7272

But we didn't add our own custom functionality. So we don't necessarily need to test it directly, it comes from Active Merchant. And hopefully we will delete that class when we upgrade the gem. It's much more important to test our integration with Stripe through our libs, services and models.

File: spec/lib/stripe/payment_intent_validator_spec.rb [hunk 1]
```diff
@@ -60,11 +60,22 @@
...
+            expect(Stripe::PaymentIntent.retrieve(
+              payment_intent.id
+            ).status).to eq("requires_capture")
+
+            Stripe::PaymentIntent.capture(payment_intent.id)
+
+            expect(Stripe::PaymentIntent.retrieve(
+              payment_intent.id
+            ).status).to eq("succeeded")
+          end
```
Comment (mkllnk) [inline spec/lib/stripe/payment_intent_validator_spec.rb:74]: This spec should be testing the Stripe::PaymentIntentValidator which is our class. The spec before this is calling the validator and checking that it doesn't raise errors but this test case is not calling it at all. As you said in your comment, it's testing Stripe directly and a similar test is probably performed by the Stripe team already. So we don't have to test this.

**Review by mkllnk — APPROVED** [review]
Nice to see this all working.

I think that there's an unnecessary test case but it probably doesn't hurt to keep it for now.

**Pull request merged**